frontends/web/src/routes/account/summary/chart.tsx

```diff
@@ -49,120 +43,176 @@ type State = {
   toolTipTop: number;
   toolTipLeft: number;
   toolTipTime: number;
-  isMobile: boolean;
 }
 
-type Props = ChartProps & TranslateProps;
+const defaultData: ISummary = {
+  chartDataMissing: true,
+  chartDataDaily: [],
+  chartDataHourly: [],
+  chartFiat: 'USD',
+  chartTotal: null,
+  formattedChartTotal: null,
+  chartIsUpToDate: false,
+  lastTimestamp: 0,
+};
 
 type FormattedData = {
   [key: number]: string;
 }
-class Chart extends Component<Props, State> {
-  static contextType = AppContext;
-  context!: React.ContextType<typeof AppContext>;
-  private ref = createRef<HTMLDivElement>();
-  private refToolTip = createRef<HTMLSpanElement>();
-  private chart?: IChartApi;
-  private lineSeries?: ISeriesApi<'Area'>;
-  private height: number = 300;
-  private mobileHeight: number = 150;
-  private formattedData?: FormattedData;
-
-  static readonly defaultProps: ChartProps = {
-    data: {
-      chartDataMissing: true,
-      chartDataDaily: [],
-      chartDataHourly: [],
-      chartFiat: 'USD',
-      chartTotal: null,
-      formattedChartTotal: null,
-      chartIsUpToDate: false,
-      lastTimestamp: 0,
-    },
-    hideAmounts: false,
-  };
-
-  public readonly state: State = {
+
+function renderDate(date: number, lang: string, src: string) {
+  return new Date(date).toLocaleString(
+    lang,
+    {
+      year: 'numeric',
+      month: '2-digit',
+      day: '2-digit',
+      ...(src === 'hourly' ? {
+        hour: '2-digit',
+        minute: '2-digit',
+      } : null)
+    }
+  );
+}
+
+export const Chart = ({ data = defaultData, noDataPlaceholder, hideAmounts = false }: TProps) => {
+  const height: number = 300;
+  const mobileHeight: number = 150;
+  const isMobile = window.innerWidth <= 640;
+  const hasData = data.chartDataDaily && data.chartDataDaily.length > 0;
+  const hasHourlyData = data.chartDataHourly && data.chartDataHourly.length > 0;
+
+  const { t, i18n } = useTranslation();
+  const { chartDisplay, setChartDisplay } = useContext(AppContext);
+
+  const ref = useRef<HTMLDivElement>(null);
+  const refToolTip = useRef<HTMLSpanElement>(null);
+  const chart = useRef<IChartApi>();
+  const lineSeries = useRef<ISeriesApi<'Area'>>();
+  const formattedData = useRef<FormattedData>({});
+
+  const [state, setState] = useState<State>({
     source: 'daily',
     toolTipVisible: false,
     toolTipValue: undefined,
     toolTipTop: 0,
     toolTipLeft: 0,
     toolTipTime: 0,
-    isMobile: false,
-  };
-
-  public componentDidMount() {
-    this.createChart();
+  });
```

Comment:
split into source/setSource, difference/setDifference, diffSince/setDiffSince and the rest tooltip/setTooltip ?